File: quiz_BTECH1050723.cpp
#include <bits/stdc++.h>
#include <vector>
using namespace std;

void dfs(vector<int>& ans, vector<int> adj[], vector<int>& vis, int start) {
        for (auto itr : adj[start]) {
            if (vis[itr] == 0) {
                ans.push_back(itr);
                vis[itr] = 1;
                dfs(ans,adj,vis,itr);
            }
        }
}

int main() {
    int n, m;
    cin >> n >> m;

    vector<int> adj[n + 1];
    for (int i = 0; i < m; i++) {
        int u, v;
        cin >> u >> v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    vector<int> ans;
    vector<int> vis(n+1, 0);
    vis[1] = 1;
    ans.push_back(1);
    
    dfs(ans, adj, vis, 1);

    cout << "DFS traversal starting from vertex 1: ";
    for (int i = 0; i < n; i++) {
        cout << ans[i] << " ";
    }
    cout << endl;

    return 0;
}

File: quiz_BTECH1050723.py
import numpy as np
def dfs(adj,n,traversal,vis,node):
    vis[node]=1
    traversal.append(node+1)
    for i in adj[node]:
        if(vis[i]==0):
            dfs(adj,n,traversal,vis,i)


def main():
    n=int(input("Enter the number of nodes:"))
    m=int(input("Enter the number of edges which are connected:"))
    adj_ls=[[] for _ in range(n)]
    for i in range(m):
        n1=int(input("Enter the first node :  "))
        n2=int(input("Enter the second node :  "))
        adj_ls[n1].append(n2)
    
    # print(adj_ls)
    traversal=[]
    vis=np.zeros(n,dtype='int')
    for i in range(n):
        dfs(adj_ls,n,traversal,vis,i)
        print(traversal)
        for i in range(n):
           vis[i]=0
        traversal.clear()
main()